homework34/src/main/java/hw34/CartController.java

```diff
@@ -0,0 +1,89 @@
+package hw34;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Scanner;
+
+@Component
+@Scope("prototype")
+public class CartController {
+    private final ProductRepository productRepository;
+    private final Cart cart;
+
+    @Autowired
+    public CartController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+        this.cart = new Cart();
```

Comment:
Object `Cart` must be provided by Spring (this is the basics of Spring and its IoC and DI). If you have, in your application, something that looks like a piece of business logic - with 99.99999% this piece of logic must be packed into a separate class and configured as a bean.